### PR TITLE
More improvements for the basic course

### PR DESCRIPTION
**Pull request by bernhardmgruber**
I went through the first day recording from last year's course and here are a few more improvements for the basic chapter.

### CODE REVIEW

File: talk/basicconcepts/headersinterfaces.tex [hunk 1]
```diff
@@ -29,21 +29,47 @@
...
-    #ifdef USE64BITS
-      typedef uint64_t myint;
-    #else
-      typedef uint32_t myint;
+    #define CHECK_ERROR(x) if ((x) != MY_GOLDEN_NUMBER) \
+      std::cerr << #x " was not the golden number\n";
+    // compile time or platform specific configuration
+    #if defined(USE64BITS) || defined(__GNUG__)
+      using myint = uint64_t;
+    #else if
```
Comment (hageboeck) [inline talk/basicconcepts/headersinterfaces.tex:39]:
```suggestion
    #elif
```

**Review by hageboeck — APPROVED** [review]
Very good, but please fix the `#elif`!

File: talk/basicconcepts/coresyntax.tex [hunk 2]
```diff
@@ -23,20 +23,20 @@
...
 
     /* multiline comment
      * in case we need to say more
      */
-    double d;
+    double /* or something in between */ d;
 
     /**
      * Best choice : doxygen compatible comments
-     * \fn bool isOdd(int i)
```
[the remark this answers is not in the document]
Comment (hageboeck) [reply talk/basicconcepts/coresyntax.tex:36]: You need it if you are documenting a function in a place for away from the actual function. Its use is very much discouraged.

File: talk/basicconcepts/control.tex [hunk 3]
```diff
@@ -193,21 +191,21 @@
...
   \end{alertblock}
 \end{frame}
 
 \begin{frame}[fragile]
-  \frametitlecpp[11]{Range based loops}
+  \frametitlecpp[11]{Range-based loops}
   \begin{block}{Reason of being}
     \begin{itemize}
-    \item simplifies loops tremendously
+    \item simplifies loops over "ranges" tremendously
```
[the remark this answers is not in the document]
Comment (bernhardmgruber) [reply talk/basicconcepts/control.tex:202]: That's useful, thx!

File: talk/basicconcepts/headersinterfaces.tex (hunk 1 above)
Comment (bernhardmgruber) [reply talk/basicconcepts/headersinterfaces.tex:39]: thx for spotting this one!

File: talk/basicconcepts/control.tex [hunk 4]
```diff
@@ -248,16 +245,17 @@
...
+  \frametitlecpp[98]{Control structures : jump statements}
+  \begin{block}{}
     \begin{description}
-    \item[break] goes out of the loop
-    \item[continue] goes immediately to next iteration
-    \item[return] goes out of current function
+    \item[break] exits the loop and continues after it
+    \item[continue] goes immediately to next loop iteration
+    \item[return] exists the current function
+    \item[goto] can jump anywhere inside a function, don't use !
```
[the remark this answers is not in the document]
Comment (hageboeck) [reply talk/basicconcepts/control.tex:254]: I'm not a friend of `goto`, but with "don't use" I'm fine.

[the remark this answers is not in the document]
Comment (bernhardmgruber) [reply talk/basicconcepts/control.tex:254]: OK, thx for the input!

File: talk/basicconcepts/coresyntax.tex (hunk 2 above)
[the remark this answers is not in the document]
Comment (bernhardmgruber) [reply talk/basicconcepts/coresyntax.tex:36]: I can see that, since the danger of documentation and code drifting apart is too big. Thx!

**Pull request merged**